### PR TITLE
Test for default-ports feature and simplify the function

### PR DESCRIPTION
**Pull request by folbricht**

### CODE REVIEW

File: cmd/routedns/main.go [hunk 1]
```diff
@@ -229,9 +229,9 @@ func start(opt options, args []string) error {
 			listeners = append(listeners, ln)
 		case "doh":
 			if l.Transport != "quic" {
-				l.Address = rdns.AddressWithDefaultForHttp(l.Address, rdns.DoHPort)
+				l.Address = rdns.AddressWithDefault(l.Address, rdns.DoHPort)
 			} else if l.Transport == "quic" {
-				l.Address = rdns.AddressWithDefaultForHttp(l.Address, rdns.DohQuicPort)
+				l.Address = rdns.AddressWithDefault(l.Address, rdns.DohQuicPort)
 			}
 			tlsConfig, err := rdns.TLSServerConfig(l.CA, l.ServerCrt, l.ServerKey, l.MutualTLS)
```
Comment (charlieporth1) [inline cmd/routedns/main.go:236]: Is this supposed be a change from `AddressWithDefaultForHttp` to `AddressWithDefault` or is that out of date?

File: cmd/routedns/resolver.go [hunk 2]
```diff
@@ -61,7 +62,7 @@ func instantiateResolver(id string, r resolver, resolvers map[string]rdns.Resolv
 			return err
 		}
 	case "doh":
-		r.Address = rdns.AddressWithDefaultForHttp(r.Address, rdns.DoHPort)
+		r.Address = rdns.AddressWithDefault(r.Address, rdns.DoHPort)
```
Comment (charlieporth1) [inline cmd/routedns/resolver.go:66]: Same here

File: cmd/routedns/main.go (hunk 1 above)
Comment (folbricht) [reply cmd/routedns/main.go:236]: It's intentional. I changed the function to support both formats, http and regular endpoints. That way you can use the same for both cases. Also added a few tests if you want to confirm this does handle both.

Comment (charlieporth1) [reply cmd/routedns/main.go:236]: Ok

**Pull request merged**